qt/scientific_interfaces/Indirect/AbsorptionCorrections.cpp

```diff
@@ -502,14 +502,19 @@ void AbsorptionCorrections::convertSpectrumAxes(
 
 /**
  * Handle completion of the absorption correction algorithm.
- *
  * @param error True if algorithm has failed.
  */
 void AbsorptionCorrections::algorithmComplete(bool error) {
   setRunIsRunning(false);
-  if (!error) {
+  // The m_saveAlgRunning flag is queried here so the
+  // processWavelengthWorkspace isn't executed at the end of the
+  // saveAlg completing, as this will throw an exception.
+  if (!error && !m_saveAlgRunning) {
     processWavelengthWorkspace();
     setOutputPlotOptionsWorkspaces({m_pythonExportWsName});
+  } else if (!error && m_saveAlgRunning) {
+    setButtonsEnabled(true);
+    m_saveAlgRunning = false;
   } else {
```

Comment:
Should `m_saveAlgRunning` be set to false in this case also? Seems like a `true` state could propagate to the next run if an error occurred